dbms/src/Interpreters/ExternalLoader.cpp

```diff
@@ -919,7 +918,8 @@ class ExternalLoader::LoadingDispatcher : private boost::noncopyable
             /// Use `create_function` to perform the actual loading.
             /// It's much better to do it with `mutex` unlocked because the loading can take a lot of time
             /// and require access to other objects.
-            auto [new_object, new_exception] = loadOneObject(name, info->object_config, info->config_changed, info->object);
+            bool need_complete_loading = !info->object || info->config_changed || info->forced_to_reload;
+            auto [new_object, new_exception] = loadOneObject(name, info->object_config, need_complete_loading ? nullptr : info->object);
```

Comment:
I think we just have to rename flag to something like `need_reload = true`. And use it like
```
if (!need_reload && previous_version)
    return previous_version->clone();
```
Now it seems like you overloaded object with flag meaning.